node/pkg/processor/reactor/reactor.go

```diff
@@ -0,0 +1,491 @@
+package reactor
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/certusone/wormhole/node/pkg/supervisor"
+
+	node_common "github.com/certusone/wormhole/node/pkg/common"
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"go.uber.org/zap"
+
+	gossipv1 "github.com/certusone/wormhole/node/pkg/proto/gossip/v1"
+	"github.com/wormhole-foundation/wormhole/sdk/vaa"
+)
+
+// ConsensusReactor implements the full consensus processor for a single Observation. It cannot be reused after being
+// finalized.
+type ConsensusReactor[K Observation] struct {
+	// group is the name of the reactor group
+	group string
+	// Current state of the reactor
+	state      State
+	stateMutex sync.Mutex
```

Comment:
This field is called `stateMutex` but it appears to be protecting many other fields in the struct.  I think it might be a bit clearer if turn `State` into a full struct and put all the mutable fields in there and only keep the immutable fields in the outer `ConsensusReactor`.  Then we can even embed the mutex in the `State` struct so that it's very clear exactly what the mutex is protecting.

Reply:
Moved to state